topic/topic.go

```diff
@@ -137,6 +138,21 @@ func (t *topic) RemoveConsumerService(value services.ServiceID) (Topic, error) {
 	return nil, fmt.Errorf("could not find consumer service %s in the topic", value.String())
 }
 
+func (t *topic) UpdateConsumerService(value ConsumerService) (Topic, error) {
+	css := t.ConsumerServices()
+	for i, cs := range css {
+		if cs.ServiceID().String() != value.ServiceID().String() {
```

Comment:
Maybe you can define an `Equal` method on the `ConsumerService` type.

Reply:
sg